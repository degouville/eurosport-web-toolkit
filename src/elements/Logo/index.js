import styled, { css } from 'react-emotion';
import React from 'react';
import PropTypes from 'prop-types';

import * as breakpoints from '../../breakpoints';
import eurosport from '../../assets/eurosport.svg';
import eurosportSmall from '../../assets/eurosport.small.svg';

const StyledLogo = styled.img`
  box-sizing: border-box;
  width: 147px;
  height: 17px;

  ${breakpoints.medium(css`
    width: 213px;
    height: 26px;
  `)};

  ${props =>
    props.small &&
    css`
      width: 20px;
      height: 23px;
    `}
`;

const Logo = ({ small }) => <StyledLogo src={small ? eurosportSmall : eurosport} alt="Eurosport" />;

Logo.css = StyledLogo;

Logo.defaultProps = {
  small: false,
};

Logo.propTypes = {
  small: PropTypes.bool,
};

export default Logo;
