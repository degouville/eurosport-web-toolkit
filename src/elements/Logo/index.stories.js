import React from 'react';
import { withInfo } from '@storybook/addon-info';
import { boolean } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import { Logo } from '../..';

storiesOf('Logo', module).add('configurable', withInfo()(() => <Logo small={boolean('small', true)} />));
