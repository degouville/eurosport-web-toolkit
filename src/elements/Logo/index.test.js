import { shallow } from 'enzyme/build';
import React from 'react';
import Logo from '.';

describe('Logo', () => {
  it('renders usual Logo by default', () => {
    expect(shallow(<Logo />)).toMatchSnapshot();
  });

  it('renders small Logo', () => {
    expect(shallow(<Logo small />)).toMatchSnapshot();
  });
});
