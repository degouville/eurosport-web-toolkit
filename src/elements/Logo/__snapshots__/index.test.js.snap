// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders small Logo 1`] = `
<Styled(img)
  alt="Eurosport"
  src="eurosport.small.svg"
/>
`;

exports[`Logo renders usual Logo by default 1`] = `
<Styled(img)
  alt="Eurosport"
  src="eurosport.svg"
/>
`;
